src/engine/sync/synccontrol.cpp

```diff
@@ -355,7 +358,15 @@ void SyncControl::trackBeatsUpdated(mixxx::BeatsPointer pBeats) {
 }
 
 void SyncControl::slotControlPlay(double play) {
-    m_pEngineSync->notifyPlaying(this, play > 0.0);
+    bool playing = play > 0.0;
+
+    // Only notify engine sync when the play state has actually changed to
+    // avoid re-syncing when the user presses play while holding the cue
+    // button.
+    if (m_bPlaying != playing) {
```

Comment:
I am not sure if this actually already is the case, but slotControlPlay can be called from different threads, since it is direct connection. So we need to use an atomic flag.
QAtomicInteger<bool> m_bPlaying
m_bPlaying.fetchAndStoreAcquire(playing);